### DISCUSSION TITLE
Cannot set value on ChoiceField

### DISCUSSION THREAD

**Discussion by tg-digineo** [gettalong/hexapdf#287] [Q&A] [answered]
Hi,
i'm trying to set the value on a ChoiceField, but i get an Error claiming that the value is invalid. Also the `#option_items`-method returns an empty array but in the inspect-output i can see the options ( :Opt=>["Herr", "Frau", "(ohne)"]). Do you have any idea what might be causing this?

```
[21] pry(#<>)> field                                                                               
=> #<HexaPDF::Type::AcroForm::ChoiceField [1037, 0] value={:F=>4, :FT=>:Ch, :Ff=>4325376, :Kids=>#<HexaPDF::PDFArray [0, 0] value=[#<HexaPDF::Type::Annotations::Widget [1124, 0] value={:AP=>{:N=>#<HexaPDF::Reference [1086, 0]>}, :BS=>{:S=>:S, :W=>1}, :DA=>"/ArialMT 10 Tf 0 g", :F=>4, :Ff=>4325376, :MK=>#<HexaPDF::Reference [1156, 0]>, :Opt=>["Herr", "Frau", "(ohne)"], :P=>#<HexaPDF::Reference [1078, 0]>, :Parent=>#<HexaPDF::Reference [1037, 0]>, :Rect=>[56.6929, 445.879, 170.079, 465.855], :Subtype=>:Widget, :Type=>:Annot}>, #<HexaPDF::Type::Annotations::Widget [632, 0] value={:AP=>{:N=>#<HexaPDF::Reference [154, 0]>}, :BS=>{:S=>:S, :W=>1}, :DA=>"/ArialMT 10 Tf 0 g", :F=>4, :FT=>:Ch, :Ff=>4325376, :MK=>#<HexaPDF::Reference [650, 0]>, :Opt=>["Herr", "Frau", "(ohne)"], :P=>#<HexaPDF::Reference [7, 0]>, :Parent=>#<HexaPDF::Reference [1037, 0]>, :Rect=>[56.6929, 647.829, 170.079, 667.805], :Subtype=>:Widget, :Type=>:Annot}>]>, :T=>"TN_Anrede"}>
[22] pry(#<>)> field.option_items                                                                  
=> []
[23] pry(#<>)> field.field_value = "Herr"                                                          
HexaPDF::Error: Invalid value "Herr" for combo_box field TN_Anrede
```

**Comment by gettalong** [comment]
Since the `#option_items` methods returns an empty array, setting a value will not work because the value is tested against the option items array.

I'm not sure, though, why `#option_items` returns an empty array in your case. Can you please provide the PDF for inspection? Thanks.

**Reply by gettalong** [reply, marked as answer]
Thanks for sending the PDF!

The reason why this doesn't work is that the field has multiple widget annotations and the /Opt key is not set on the field itself but on the widget annotations.

I'm not sure if this is standards conform. And there is also the problem if the /Opt keys of multiple widgets don't agree on the allowed values.

To circumvent this problem, I will add a provision to HexaPDF to move the /Opt key of one of the widgets to the field itself and then delete all /Opt keys from the widgets.
